libs/@local/repo-chores/scripts/check-license-in-workspaces.ts

```diff
@@ -0,0 +1,139 @@
+import path from "node:path";
+
+import chalk from "chalk";
+import execa from "execa";
+import globby from "globby";
+
+type WorkspaceInfo = {
+  location: string;
+  workspaceDependencies: string[];
+  mismatchedWorkspaceDependencies: string[];
+};
+
+const monorepoRootDirPath = path.resolve(__filename, "../../../../..");
+
+const script = async () => {
+  console.log("Checking license files in Yarn workspaces...");
+
+  const { stdout } = await execa("yarn", ["--silent", "workspaces", "info"], {
+    env: { PATH: process.env.PATH },
+    extendEnv: false, // Avoid passing FORCE_COLOR to a sub-process
+  });
+
+  const workspaceInfoLookup = JSON.parse(stdout) as Record<
+    string,
+    WorkspaceInfo
+  >;
+
+  const workspaceDirPaths = [
+    monorepoRootDirPath,
+    ...Object.entries(workspaceInfoLookup).map(([, workspaceInfo]) =>
+      path.resolve(monorepoRootDirPath, workspaceInfo.location),
+    ),
+  ];
+
+  const licenseFilePaths = await globby("**/license*", {
+    absolute: true,
+    caseSensitiveMatch: false,
+    cwd: monorepoRootDirPath,
+    ignore: ["**/node_modules/**", "**/apps/engine/*/**/*"],
```

Comment:
Why is `engine` ignored?

Reply:
I clarified the reason in https://github.com/hashintel/hash/pull/1825/commits/f3fd69842a32dc974e7a4282674686781931f199. If we don't have this glob, we get this in output:

```
[EXTRA]    apps/engine/lib/memory/src/arrow/util/LICENSE
[EXTRA]    apps/engine/lib/memory/src/arrow/ipc/serialize/LICENSE
```

After some more thinking, I removed this glob. Lines marked as `[EXTRA]` don’t fail CI. https://github.com/hashintel/hash/pull/1825/commits/94a0ec1c3ef04650f6c4643dafc3683d783e14a7